### ISSUE TITLE
Emojis not saving 

### ISSUE THREAD

**Issue by revvin76** [opensource-socialnetwork/opensource-socialnetwork#1272]
Hi there
I am having trouble with the emojis saving to wall posts or messages.
The object in the database is just showing an empty value.

Database: Amazon AWS RDS (MySQL)
Hosted on an AWS EC2 Instance.
OSSN Version 4.6

I have disabled all my custom plugins, even copied the components from the installation file just in case i'd stupidly made any changes... also copied over the actions/classes/libraries and system folders from the installation source for good measure. 

I have reverted to GoBlue 4.4 theme to completely rule out any local files - but it still doesn't work?

I'm starting to think it may be the Database configuration? I've tried changing all the tables to UTF8MB4-General_CI and UTF8_bin but neither fixed it. Although, when I changed everything to UTF8MB4, the emoji appeared in the object table as ???? instead of the normal blank value.

Any pointers would be appreciated - as I think this is more a setup/pre-req issue than anything local.

Any other information you need, please do not hesitate to ask.

Kev
![image](https://user-images.githubusercontent.com/32345166/36987774-1824a98e-2095-11e8-9896-e6fadfdeb15a.png)

**Comment by revvin76** [comment]
I have just fixed it temporarily by following the advice in https://github.com/opensource-socialnetwork/opensource-socialnetwork/issues/1123

My whole database is now UTF8MB4_Unicode

And I changed line (78) in OssnDatabase.php to "utf8mb4" and it all started working instantly. 

I really don't like changing core files, so is this my database being wrongly configured, or does there need to be some code in the core to detect and use the correct UTF type?

Thanks again
Kev

**Comment by lianglee** [comment]
@Lablnet can you check this please?

**Comment by Lablnet** [comment]
@lianglee  ofcouse isir ts work fine in post as well as message
![screenshot 86](https://user-images.githubusercontent.com/27757785/37243758-900d16d4-24a0-11e8-969e-c874ba9d2f75.png)

**Comment by Lablnet** [comment]
As well working in my live ossn based site
http://malikumer.felahatec.com/ossn/ossn/

**Comment by lianglee** [comment]
@Lablnet  Thanks Malik, 
@revvin76 Closing it as works for us , seems issue with database server.

**Comment by revvin76** [comment]
Ok.
The database you were testing on - was that UTF8MB4 or just UTF8 ?
Any suggestions to why changing that line in the core code fixes the issue? :(

**Comment by lianglee** [comment]
The tables are utf8_general_ci,   can you try on development release?

**Comment by revvin76** [comment]
I've manually changed all my table collations back to utf8_general_ci and i've upgraded the site to the dev version - and not made any modifications - and its all working fine. Old messages are still displaying properly and new ones store the emojis fine. 

So it seems the issue has just disappeared?

My Ossn still says v4.6 even though I installed the DEV from 5.0 Repo? But I guess thats not really a problem?

Any way I appreciate your time checking this out. Thanks for your assistance :)
